src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -324,7 +352,7 @@ static public void checkTypeTagInInterfacesAndExternClass(final HaxeMethodModel
       if (currentMethod.getReturnTypeTagPsi() == null) {
         holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), HaxeBundle.message("haxe.semantic.type.required"));
       }
-      for (final HaxeParameterModel param : currentMethod.getParameters()) {
+      for (final HaxeParameterModel param : currentMethod.getParameters().parameters) {
```

Comment:
I am definitely not a fan of reaching through an object to get at a public field.  I realize that you've got the field marked NotNull so that it's unlikely to be an error.  I'd much rather see an Iterable here.  Why not have the HaxeParametersModel implement the Iterable interface? So the code would revert to the old line 327.  (The same comment goes for all uses of .parameters that you have sprinkled throughout this PR.